subgraphs/uniswap-v3/src/common/entities/token.ts

```diff
@@ -54,8 +59,13 @@ export function getOrCreateToken(
     token.save();
   }
 
-  if (token.lastPriceBlockNumber != event.block.number && getNewPrice) {
-    token.lastPriceUSD = findUSDPricePerToken(event, token);
+  if (
+    event.block.number.minus(token.lastPriceBlockNumber!).gt(BIGINT_TEN) &&
```

Comment:
Not against this change, but could it happen that by leaving a 10 block margin between price updates we might be skipping blocks where we had price issues? Potentially making it look like the price controls you added work, when in reality we just skipped that block?

Have you validated that they work without this? :P

Reply:
Not as thoroughly as I have with it. I can try that out. This definitely could happen, but on this, I think more than likely it'll still catch those instances. This is because there are usually multiple blocks or many blocks between pricings on these smaller pools.

Reply:
I changed this to try to improve performance, not sure its helping too much though.